spec/betterc.dd

```diff
@@ -0,0 +1,48 @@
+Ddoc
+
+$(SPEC_S Better C,
+
+`-betterC` is a Command-Line flag for `dmd`, which generated light-weight,
+barebone, runtime-free binaries with are suitable for linking from C.
+
+---
+void main()
+{
+    import core.stdc.stdio : printf;
+    printf("Hello betterC");
+}
+---
+
+$(CONSOLE
+> dmd -betterC hello.d
+Hello betterC
+)
+
+$(H2 $(LNAME2 effects, Effects))
+
+    $(OL
+        $(LI Doesn't link DRuntime and standard library by default)
+	    $(LI Doesn't generate ModuleInfo entries)
+    )
+
+$(H2 $(LNAME2 consequences, Consequences))
+
+As no DRuntime is available, many D features won't work. For example:
+
+	$(OL
+	    $(LI No static module constructors or deconstructors)
+	    $(LI No Garbage Collection)
+	    $(LI No dynamic arrays and associative arrays)
+	    $(LI No Classes)
```

Comment:
> They still emit references to Typeinfo_Class

But that's good! How else would you initialize instances of them? At minimum, the vtbl ptr can't be initialized in the constructor by user code (easily). Until we move all of TypeInfo* to traits / templates that lazily pull only what's needed, we still need Typeinfo. 

I tried the [following](https://gist.github.com/ZombineDev/e070f4dc4a9a3d18d6b18156ca318d91) and it seems to work well.

`betterC` is a spectrum - yes if you're doing bare-metal development you really don't want those stuff, but if you're writing a plugin and you don't want to carry druntime, some of the TypeInfo stuff can still be useful. So when talking about `betterC`, I think we need to differentiate between regular user-land apps / libraries that still rely on libc, apps / libraries that don't rely on libc (doing system calls themselves) and real bare-metal development.

---
P.S.:

After playing a bit, I saw what you meant. I (wrongly) expected that the emitted type info into the binary would be self-contained. Unfortunately that's not the case. The following code:
```D
extern (C++) class C { }
extern (C) void main() { }
```
Produces a undefined reference to `_D14TypeInfo_Class6__vtblZ`:
```sh
$ dmd -c -betterC extern_cpp_class.d && nm extern_cpp_class.o
0000000000000000 t
                 U _D14TypeInfo_Class6__vtblZ
0000000000000000 V _D16extern_cpp_class1C6__initZ
0000000000000000 V _D16extern_cpp_class1C6__vtblZ
0000000000000000 V _D16extern_cpp_class1C7__ClassZ
0000000000000000 W main
```

Also while experimenting with [that](https://gist.github.com/ZombineDev/e070f4dc4a9a3d18d6b18156ca318d91) I ran into [this](https://issues.dlang.org/show_bug.cgi?id=17605) bug.

Also `extern (Windows)` is much worse than `extern (C++)` classes:
```D
extern (Windows) class C { }
extern (C) void main() { }
```
```sh
$ dmd -c -betterC test.d && nm test.o
0000000000000000 t
                 U _D14TypeInfo_Class6__vtblZ
0000000000000000 V _D16extern_cpp_class1C6__initZ
0000000000000000 V _D16extern_cpp_class1C6__vtblZ
0000000000000000 V _D16extern_cpp_class1C7__ClassZ
                 U _D6object6Object5opCmpMFC6ObjectZi
                 U _D6object6Object6toHashMFNbNeZm
                 U _D6object6Object8opEqualsMFC6ObjectZb
                 U _D6object6Object8toStringMFZAya
                 U _D6Object7__ClassZ
                 U _d_dso_registry
0000000000000000 W main
                 U __start_minfo
                 U __stop_minfo
```

---

So scratch everything I said, `extern (C++)`, or worse `extern (Windows)` classes are really not usable without linking druntime with your object files.